addons/hr_holidays/static/src/bugfix/bugfix.js

```diff
@@ -8,3 +8,80 @@ odoo.define('hr_holidays/static/src/bugfix/bugfix.js', function (require) {
 'use strict';
 
 });
+
+// FIXME move me in hr_holidays/static/src/models/partner/partner.js
+odoo.define('hr_holidays/static/src/models/partner/partner.js', function (require) {
+'use strict';
+
+const {
+    registerClassPatchModel,
+    registerFieldPatchModel,
+    registerInstancePatchModel,
+} = require('mail/static/src/model/model_core.js');
+const { attr, one2one } = require('mail/static/src/model/model_field.js');
+const { clear } = require('mail/static/src/model/model_field_command.js');
+
+const { str_to_datetime } = require('web.time');
+
+registerClassPatchModel('mail.partner', 'hr_holidays/static/src/models/partner/partner.js', {
+    /**
+     * @override
+     */
+    convertData(data) {
+        const data2 = this._super(data);
+        if ('out_of_office_date_end' in data) {
+            data2.out_of_office_date_end = data.out_of_office_date_end ? data.out_of_office_date_end : clear();
+        }
+        return data2;
+    },
+});
+
+registerInstancePatchModel('mail.partner', 'hr_holidays/static/src/models/partner/partner.js', {
+    /**
+     *
+     * @private
+     */
+    _computeOutOfOfficeText() {
+        if (!this.out_of_office_date_end) {
+            return clear();
+        }
+        if (!this.messaging.locale.language) {
+            return clear();
+        }
+        const currentDate = new Date();
+        const date = str_to_datetime(this.out_of_office_date_end);
+        const options = { day: 'numeric', month: 'short' };
+        if (currentDate.getFullYear() !== date.getFullYear()) {
+            options.year = 'numeric';
+        }
+        const localeCode = this.messaging.locale.language.replace(/_/g,'-');
+        const formattedDate = date.toLocaleDateString(localeCode, options);
+        return _.str.sprintf(this.env._t("Out of office until %s"), formattedDate);
+    },
+
+});
+
+registerFieldPatchModel('mail.partner', 'hr/static/src/models/partner/partner.js', {
+    /**
+     * Serves as compute dependency.
+     */
+    messagingLocale: one2one('mail.locale', {
+        related: 'messaging.locale',
+    }),
+    /**
+     * Date of end of the out of office period of the partner as string.
```

Comment:
Add expected format for the string as comment too. Either with an example or using Y M D etc notation